### ISSUE TITLE
Roadmap

### ISSUE THREAD

**Issue by zheng1** [kubesphere/porter#81]
## v0.2.0
- Support layer 2 loadbalancing #75 #31
- Support loadBalancerIP in Service #73

## v0.3.0
- Support add neighbor dynamically #69
- Support config porter via CRD #69

## v0.4.0
- Support non-bgp switch #79

**Comment by FeynmanZhou** [comment]
Awesome, very glad to see this roadmap.

**Comment by k0ngk0ng** [comment]
I'm interested in this roadmap and want to be a contributor. 😃

**Comment by FeynmanZhou** [comment]
> I'm interested in this roadmap and want to be a contributor. 😃

Cool! Another good news today. @k0ngk0ng is going to upgrade from user to contributor. We will set up a time to discuss the roadmap and staged tasks.

**Comment by jenningsloy318** [comment]
Is it possible to add multi master node  support, as common case we have multiple  master k8s nodes?

**Comment by duanjiong** [comment]
> Is it possible to add multi master node support, as common case we have multiple master k8s nodes?

Are you saying that porter-manager deploys multiple copies and then they can process requests simultaneously, equivalent to active-active?

**Comment by jenningsloy318** [comment]
As the BgpConf will config the `routerID` to of one k8S master node where the manager is running on, and how to determine which master node it is running on when there is not only one master node?

**Comment by duanjiong** [comment]
No, it's not. You can specify any one of them. it doesn't matter if the porter-manager switches to another node。
